app/src/main/java/org/sagebionetworks/research/mpower/medication/MedicationDayFragment.kt

```diff
@@ -0,0 +1,175 @@
+            args.putString(ARG_DAYS, days)
+            fragment.arguments = args
+            return fragment
+        }
+    }
+
+    lateinit var customView: View
+    lateinit var listener: DaySelectedListener
+    var selectedDays: MutableList<String> = arrayListOf()
+
+    lateinit var schedId: String
+    lateinit var name: String
+    lateinit var time: String
+    lateinit var days: String
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if(getArguments() != null) {
+            schedId = arguments!!.getString(ARG_SCHED_ID)
+            name = arguments!!.getString(ARG_NAME)
+            time = arguments!!.getString(ARG_TIME)
+            days = arguments!!.getString(ARG_DAYS)
+
+            selectedDays = days.split(",").toMutableList()
+            val size = selectedDays.size
+            LOGGER.debug("Selected days on init: $size")
+        } else {
+            LOGGER.debug("No arguments found")
+            name = "Default"
+            time = "9:15 PM"
+        }
+
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?): View?
+    {
+        LOGGER.debug("onCreateView()")
+        return customView
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        LOGGER.debug("onCreateDialog()")
+
+        var inflater: LayoutInflater = activity!!.layoutInflater
+        customView = inflater.inflate(R.layout.dialog_medication_day, null)
+
+        return AlertDialog.Builder(context!!)
+
+                .setView(customView)
+                .create()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        LOGGER.debug("onViewCreated()")
+        
+        day_selection_title.text = getString(R.string.medication_day_selection_title, name, time)
+        var recycler = medication_day_recycler
+        recycler.layoutManager = LinearLayoutManager(context)
+        recycler.adapter = DayAdapter(getDays(), context!!)
+        recycler.addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL))
+
+        day_selection_back.setOnClickListener {
+            _ -> dismiss()
+        }
+
+        day_selection_save.setOnClickListener {
+            _ ->
+                listener.onDaySelected(schedId, selectedDays.joinToString(","))
+                dismiss()
+        }
+
+
+    }
+
+    override fun onAttach(activity: Activity) {
+        super.onAttach(activity)
+        try {
+            listener = activity as DaySelectedListener
+        } catch (e: ClassCastException) {
+            throw ClassCastException(activity.toString() + " must implement DaySelectedListener")
+        }
+    }
+
+    fun getDays(): ArrayList<String> {
+        return arrayListOf("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday")
```

Comment:
I think these should potentially be String references to support other languages if needed.